android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -347,7 +347,10 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
 		} else if (key.equals(TiC.PROPERTY_COLOR)) {
 			tv.setTextColor(TiConvert.toColor((String) newValue));
 		} else if (key.equals(TiC.PROPERTY_HINT_TEXT)) {
-			int type = proxy.getProperties().getInt(TiC.PROPERTY_HINT_TYPE);
+			int type = UIModule.HINT_TYPE_STATIC;
+			if (proxy.getProperties() != null) {
```

Comment:
assign to a temp var to avoid grabbing it via getter twice?

Also, we should consider using a null object pattern here to avoid the null check altogether. If a KrollProxy has a null `properties` field, can't we return an empty KrollDict and get the same behavior typically? (i.e. returns null for everything, returns default values if passed in)

Reply:
I was going to change this to use `proxy.getProperty(TiC.PROPERTY_HINT_TYPE)` instead which will return `null` if the properties dictionary is `null`, but I can see this PR is already merged. That said, I think the current code is fine anyways.